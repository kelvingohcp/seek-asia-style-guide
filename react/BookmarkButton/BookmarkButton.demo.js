import React from 'react';
import PropTypes from 'prop-types';
import { BookmarkButton } from 'seek-asia-style-guide/react';
import * as sketch from './BookmarkButton.sketch';

const Container = ({ component: DemoComponent, componentProps }) => (
  <DemoComponent {...componentProps} />
);

Container.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

export default {
  route: '/bookmark-button',
  title: 'BookmarkButton',
  category: 'Components',
  component: BookmarkButton,
  container: Container,
  sketch,
  initialProps: {
    isBookmarked: false,
    onBookmarkClick: () => {}
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Bookmarked',
          transformProps: ({ ...props }) => ({
            ...props,
            isBookmarked: true
          })
        }
      ]
    }
  ]
};
