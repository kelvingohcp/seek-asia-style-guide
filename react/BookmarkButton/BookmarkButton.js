import React from 'react';
import PropTypes from 'prop-types';
import BookmarkIcon from '../BookmarkIcon/BookmarkIcon';
import Button from '../Button/Button';
import classnames from 'classnames';
import styles from './BookmarkButton.less';
export default class BookmarkButton extends React.Component {
  constructor() {
    super();

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    const { onBookmarkClick } = this.props;
    e.preventDefault();
    e.stopPropagation();
    onBookmarkClick();
  }

  render() {
    const { isBookmarked, className, ...restProps } = this.props;
    return (
      <Button color="hyperlink" className={classnames(styles.button, className, { [styles.bookmarkFilled]: isBookmarked })} onClick={this.handleClick} >
        <BookmarkIcon isBookmarked={isBookmarked} {...restProps} />
      </Button>
    );
  }
}

BookmarkButton.displayName = 'BookmarkButton';

BookmarkButton.propTypes = {
  onBookmarkClick: PropTypes.func.isRequired,
  isBookmarked: PropTypes.bool,
  className: PropTypes.string
};
