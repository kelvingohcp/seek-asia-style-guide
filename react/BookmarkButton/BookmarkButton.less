@import (reference) '~seek-asia-style-guide/theme';

.button {
  color: @saGrey4;
  &:focus {
    outline: none;
  }
  &:hover {
    color: @saGrey3;
  }
  .filled {
    &:hover {
      color: @saBlue1;
    }
  }
}

.bookmarkFilled {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-name: bounce;
  animation-name: bounce;
}

@keyframes bounce {
  from,
  to {
    -webkit-animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

  40%,
  50% {
    -webkit-animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
    animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
    -webkit-transform: translate3d(0, -6px, 0);
    transform: translate3d(0, -6px, 0);
  }
}
