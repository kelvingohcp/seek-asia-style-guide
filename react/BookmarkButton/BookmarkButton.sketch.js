import React from 'react';
import mapKeys from 'lodash/mapKeys';
import BookmarkButton from './BookmarkButton';
export const listItem = {
  'Not bookmarked': <BookmarkButton onBookmarkClick={() => {}} />,
  'Bookmarked': <BookmarkButton isBookmarked={true} onBookmarkClick={() => {}} />
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(listItem, (value, key) => `Badge/${key}`);
