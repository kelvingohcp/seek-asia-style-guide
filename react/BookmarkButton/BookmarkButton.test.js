import React from 'react';
import { shallow } from 'enzyme';
import BookmarkButton from './BookmarkButton';

const defaultProps = {
  onBookmarkClick: () => {}
};

describe('BookmarkButton', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<BookmarkButton {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render bookmarked', () => {
    const testProps = {
      ...defaultProps,
      isBookmarked: true
    };
    const wrapper = shallow(<BookmarkButton {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with class', () => {
    const testProps = {
      ...defaultProps,
      className: 'aClassyClass'
    };
    const wrapper = shallow(<BookmarkButton {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
