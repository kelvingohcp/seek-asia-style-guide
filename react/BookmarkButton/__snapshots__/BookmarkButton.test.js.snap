// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkButton should render bookmarked 1`] = `
<Button
  className="button bookmarkFilled"
  color="hyperlink"
  component="button"
  onClick={[Function]}
>
  <BookmarkIcon
    isBookmarked={true}
    onBookmarkClick={[Function]}
  />
</Button>
`;

exports[`BookmarkButton should render with class 1`] = `
<Button
  className="button aClassyClass"
  color="hyperlink"
  component="button"
  onClick={[Function]}
>
  <BookmarkIcon
    onBookmarkClick={[Function]}
  />
</Button>
`;

exports[`BookmarkButton should render with default props 1`] = `
<Button
  className="button"
  color="hyperlink"
  component="button"
  onClick={[Function]}
>
  <BookmarkIcon
    onBookmarkClick={[Function]}
  />
</Button>
`;
